charts/consul/templates/server-acl-init-job.yaml

```diff
@@ -129,6 +131,13 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: metadata.name
+        # Extract the Vault namespace from the Vault agent annotations.
+        {{- if .Values.global.secretsBackend.vault.enabled }}
+        {{- if .Values.global.secretsBackend.vault.agentAnnotations }}
+        - name: VAULT_NAMESPACE
+          value: {{ get (tpl .Values.global.secretsBackend.vault.agentAnnotations . | fromYaml) "vault.hashicorp.com/namespace" }}
```

Comment:
This ended up being semi-ugly. To specify a Vault namespace, you add a `vault.hashicorp.com/namespace` annotation. When the Consul helm chart, this is specified in the "extra" `global.secretsBackend.vault.agentAnnotations`. So this parses the namespace from that.

When talking through the Vault agent, even though it uses the namespace for injected secrets/templates, the agent doesn't scope API requests to that namespace I guess. So this needs to be explicitly specified for the Vault client used by server-acl-init when reading/writing secrets in the Vault API.